monkey/agent_plugins/exploiters/wmi/src/wmi_client.py

```diff
@@ -81,6 +84,37 @@ def login(self, host: TargetHost, credentials: Credentials):
 
         self._connected = True
 
+    def _dcom_firewall_checker(self, iInterface, timeout: float):
+        stringBinding = None
+        stringBindings = iInterface.get_cinstance().get_string_bindings()
+        for strBinding in stringBindings:
+            if strBinding["wTowerId"] == 7:
+                if strBinding["aNetworkAddr"].find("[") >= 0:
+                    binding, _, bindingPort = strBinding["aNetworkAddr"].partition("[")
+                    bindingPort = "[" + bindingPort
+                else:
+                    binding = strBinding["aNetworkAddr"]
+                    bindingPort = ""
+
+                if binding.upper().find(iInterface.get_target().upper()) >= 0:
+                    stringBinding = "ncacn_ip_tcp:" + strBinding["aNetworkAddr"][:-1]
+                    break
+                elif (
+                    iInterface.is_fqdn()
+                    and binding.upper().find(iInterface.get_target().upper().partition(".")[0]) >= 0
+                ):
+                    stringBinding = f"ncacn_ip_tcp:{iInterface.get_target()}{bindingPort}"
+        if stringBinding is None:
+            raise Exception("Exception occured defining string binding")
+        try:
+            rpctransport = transport.DCERPCTransportFactory(stringBinding)
+            rpctransport.set_connect_timeout(timeout)
+            rpctransport.connect()
+            rpctransport.disconnect()
```

Comment:
So this is essentially establishing another connection and then immediately disconnecting? Isn't there another way to add a timeout?

Reply:
Not that I found.